code/modules/surgery/organs/organ_internal.dm

```diff
@@ -151,8 +151,7 @@
 		S.tastes["meat"] = 3
 
 	if(organ_flags & ORGAN_EDIBLE)
-		var/datum/component/edible/EC = src.GetComponent(/datum/component/edible)
-		EC.checkLiked(1, M)
+		SEND_SIGNAL(src, COMSIG_EDIBLE_CHECK_LIKED, M)
```

Comment:
who made you do this, is there even any proof this is more performant than the get component call?

Reply:
i saw a line of code from goof and thought it was better for a one off minute use, my b

Reply:
@ninjanomnom answer!

Reply:
@optimumtact it's not about performance its about using components correctly. components are supposed to be self-contained. Either way this is not the way to do this because signals should be sent on the definition level of the proc, not on a subtype.

Reply:
my god he's alive